src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -724,10 +732,13 @@ class IntervalExpressionSegment(ansi.IntervalExpressionSegment):
         TypedParser("numeric_literal", ansi.LiteralSegment, type="numeric_literal"),
         Ref("ParameterizedSegment"),
     ),
-    # Add three elements to the ansi LiteralGrammar
+    QuotedLiteralSegment=OneOf(
+        Ref("SingleQuotedLiteralSegment"),
+        Ref("DoubleQuotedLiteralSegment"),
+    ),
```

Comment:
This is a very sensible thing to do. 👍 

It's a slight scope creep for this PR, but now that you've made the `QuotedLiteralSegment`, there are a bunch of other places in the BigQuery dialect which currently say `OneOf(Ref("SingleQuotedLiteralSegment"), Ref("DoubleQuotedLiteralSegment"))`. If we're adding this simplifaction, I think we should also apply it in all the other places in `dialect_bigquery.py`.

Would you be able to add that? Otherwise this looks great.